### PR TITLE
Fix #3515: Do not save tasty in .class with -YemitTasty

### LINKED ISSUE #3515: Don't emit a TASTY section in classfiles when we're also emitting .tasty files
`-YemitTasty` can be used to generate .tasty files, but when we do this we still store TASTY in classfiles, this is unnecessary duplication.

### PR DESCRIPTION
**Pull request by nicolasstucki**
Only emit .tasty file

### CODE REVIEW

**Comment by nicolasstucki** [comment]
We are currently not loading tasty from the `.tasty` files. Will need to add this functionality in this PR.

File: compiler/src/dotty/tools/dotc/core/classfile/ClassfileParser.scala [hunk 1]
```diff
@@ -781,7 +781,15 @@ class ClassfileParser(
 
       if (scan(tpnme.TASTYATTR)) {
         val attrLen = in.nextInt
-        return unpickleTASTY(in.nextBytes(attrLen))
+        if (attrLen == 0) {
+          // A tasty attribute implies the existence of the .tasty file
+          val tastyFilePath =
+            classfile.jpath.resolveSibling(classfile.jpath.getFileName.toString.dropRight("class".length) + "tasty")
```
Comment (allanrenucci) [inline compiler/src/dotty/tools/dotc/core/classfile/ClassfileParser.scala]:
```scala
io.File(classfile.jpath).changeExtension("class")
```

Comment (nicolasstucki) [reply compiler/src/dotty/tools/dotc/core/classfile/ClassfileParser.scala]: Thanks

**Comment by nicolasstucki** [comment]
We have them in the idempotency tests, that is why ed61847 failed the tests.

**Comment by nicolasstucki** [comment]
I will create one test that will execute in local tests

**Pull request merged**